### PR TITLE
Drop fixed pipeline lighting stuff

### PR DESCRIPTION
**Pull request by grorp**
Reasoning: https://github.com/minetest/minetest/pull/15144#issuecomment-2349313830

Bonus: This PR fixes broken sun/moon tonemaps on ogles2 and opengl3

## To do

This PR is a Ready for Review.

## How to test

Disable shaders, stuff shouldn't be broken.

### CODE REVIEW

File: src/client/wieldmesh.h [hunk 1]
```diff
@@ -132,9 +132,6 @@ class WieldMeshSceneNode : public scene::ISceneNode
 	scene::IMeshSceneNode *m_meshnode = nullptr;
 	video::E_MATERIAL_TYPE m_material_type;
 
-	// True if SMaterial::Lighting should be enabled.
-	bool m_lighting;
```
Comment (grorp) [inline src/client/wieldmesh.h:136]: It seems this was always false anyway

File: src/client/sky.cpp [hunk 2]
```diff
@@ -809,7 +804,6 @@ void Sky::setMoonTexture(const std::string &moon_texture,
 		m_materials[4].setTexture(0, m_moon_texture);
 		m_materials[4].MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL;
 		disableTextureFiltering(m_materials[4]);
-		m_materials[4].Lighting = !!m_moon_tonemap;
```
Comment (grorp) [inline src/client/sky.cpp:809]: The main thing left to do for this PR is to figure out what this and the rest of sky.cpp were doing with that lighting

File: src/client/sky.cpp [hunk 3]
```diff
@@ -95,7 +94,6 @@ Sky::Sky(s32 id, RenderingEngine *rendering_engine, ITextureSource *tsrc, IShade
 
 	for (int i = 5; i < 11; i++) {
 		m_materials[i] = baseMaterial();
-		m_materials[i].Lighting = true;
```
Comment (grorp) [inline src/client/sky.cpp:98]: Removing this luckily doesn't seem to have any effect (used for texture skybox rendering)

File: src/client/content_cao.cpp [hunk 4]
```diff
@@ -648,12 +647,7 @@ void GenericCAO::addToScene(ITextureSource *tsrc, scene::ISceneManager *smgr)
 
 	auto setMaterial = [this] (video::SMaterial &mat) {
 		mat.MaterialType = m_material_type;
-		mat.Lighting = false;
 		mat.FogEnable = true;
-		if (m_enable_shaders) {
-			mat.GouraudShading = false;
-			mat.NormalizeNormals = true;
-		}
```
Comment (grorp) [inline src/client/content_cao.cpp:650]: no-op since these properties are only used by the fixed pipeline

**Review by sfan5 — COMMENTED** [review]
looks good, will test later.

**Review by sfan5 — APPROVED** [review]

**Pull request merged**